src/main/java/org/folio/search/integration/ResourceChangeFilterStrategy.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.search.integration;
+
+import static org.folio.search.utils.SearchUtils.AUTHORITY_RESOURCE;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.folio.search.domain.dto.ResourceEvent;
+import org.springframework.kafka.listener.adapter.RecordFilterStrategy;
+
+@Log4j2
+public class ResourceChangeFilterStrategy implements RecordFilterStrategy<String, ResourceEvent> {
+
+  @Override
+  public boolean filter(ConsumerRecord<String, ResourceEvent> consumerRecord) {
+    log.info("Processing resource event [id: {}]", consumerRecord.value().getId());
+    var resourceEvent = consumerRecord.value();
+    var resourceName = resourceEvent.getResourceName();
+
+    var result = false;
+    if (resourceName.equals(AUTHORITY_RESOURCE)) {
+      log.info("Processing authority resource event [id: {}]", resourceEvent.getId());
+      result = checkAuthorityEventType(resourceEvent);
+    }
+    return result;
+  }
+
+  private boolean checkAuthorityEventType(ResourceEvent resourceEvent) {
+    return switch (resourceEvent.getType()) {
+      case UPDATE -> {
+        if (resourceHasChanges(resourceEvent)) {
+          yield false;
+        } else {
+          log.info("Skip message. No need to process update event for authority resource");
+          yield true;
+        }
+      }
+      case CREATE, REINDEX -> true;
+      case DELETE, DELETE_ALL -> {
```

Comment:
Not sure about the ```DELETE_ALL```, but I think both authority soft and hard delete operations would have event type ```DELETE```, the differentiation would be based on the subtype which would be encoded in a header of the event.